component/receiver.go

```diff
@@ -55,37 +55,6 @@ type LogsReceiver interface {
 	Receiver
 }
 
-// ReceiverFactoryBase defines the common functions for all receiver factories.
-type ReceiverFactoryBase interface {
```

Comment:
I think we should keep this to avoid repeating CreateDefaultConfig in 3 places. Maybe rename this to ReceiverFactory to have a nicer name.

Reply:
You cannot, CreateDefaultConfig returns different type base so cannot share between receiver/porcessor/etc. Logs will not repeat soon when I will merge the Logs factory PR.

Reply:
If you think about having one factory per type we can do that next, focus on removing the old Factories for the moment